### PR TITLE
Speedup Vignette and Graduated Filter

### PR DESCRIPTION
**Pull request by heckflosse**

### CODE REVIEW

**Comment by heckflosse** [comment]
I opened this pr almost more than 7 weeks ago. Any objections to merge into master?

File: rtengine/iptransform.cc [hunk 1]
```diff
@@ -517,22 +579,18 @@ static float calcPCVignetteFactor(const struct pcv_params& pcv, int x, int y)
...
-        b = fabs((x - pcv.x1) - pcv.w * 0.5);
-    } else {
-        a = fabs((x - pcv.x1) - pcv.w * 0.5);
-        b = fabs((y - pcv.y1) - pcv.h * 0.5);
+    if(pcv.is_portrait) {
+        std::swap(a, b);
     }
 
-    float angle = xatan2f(b, a);
     float dist = sqrtf(a * a + b * b);
```
Comment (Floessie) [inline rtengine/iptransform.cc]: Could be `normn(a, b, 2)`.

File: rtengine/iptransform.cc [hunk 2]
```diff
@@ -546,44 +604,47 @@ static float calcPCVignetteFactor(const struct pcv_params& pcv, int x, int y)
...
+        float dist_ie2 = pcv.ie2_mul * dist_oe2;
+        dist_oe = dist_oe1 * (1.f - pcv.sepmix) + dist_oe2 * pcv.sepmix;
+        dist_ie = dist_ie1 * (1.f - pcv.sepmix) + dist_ie2 * pcv.sepmix;
     } else {
         dist_oe = pcv.oe_a * pcv.oe_b / sqrtf(SQR(pcv.oe_b * sincosval.y) + SQR(pcv.oe_a * sincosval.x));
-        dist_ie = pcv.ie_mul * dist_oe * (1.0 - pcv.feather);
+        dist_ie = pcv.ie_mul * dist_oe;
     }
 
     if (dist <= dist_ie) {
```
Comment (Floessie) [inline rtengine/iptransform.cc]: Is this still true? I mean, `dist_ie` is no longer scaled by `(1.0 - pcv.feather)`, whereas `dist` seems unchanged.

File: rtengine/iptransform.cc [hunk 3]
```diff
@@ -382,21 +442,23 @@ static float calcGradientFactor(const struct grad_params& gp, int x, int y)
             float val = ((float)(gy - gp.top_edge_0) * gp.ys_inv);
 
             if (gp.bright_top) {
-                val = 1.0 - val;
+                val = 1.f - val;
             }
 
-            if (gp.scale < 1.0) {
-                val = pow(xsinf(val * M_PI / 2), 3);
+            val *= (M_PI / 2);
```
Comment (Floessie) [inline rtengine/iptransform.cc]: My `math.h` has a `M_PI_2` define. Plus, there is a `RT_PI` in this file, which could be copied as `RT_PIF` to get rid of the `double`. But maybe not worth the effort...

Comment (Floessie) [inline rtengine/iptransform.cc]: `sleef.c` has a `M_PIf_2`, BTW.

**Comment by Floessie** [comment]
@heckflosse Compiles and works fine. :+1:

File: rtengine/iptransform.cc (hunk 2 above)
Comment (heckflosse) [inline rtengine/iptransform.cc]: `pcv.ie_mul` is scaled by `(1.0 - pcv.feather)` outside the loop

**Comment by Hombre57** [comment]
Agreement to merge gave on IRC.

**Pull request merged**